src/Adapter/MailTemplate/MailPreviewVariablesBuilder.php

```diff
@@ -351,6 +375,28 @@ private function getProductList(Order $order)
         return $productTemplateList;
     }
 
+    /**
+     * @return array
+     */
+    private function getFakeVirtualProductList()
+    {
+        $products = Product::getProducts($this->context->language->getId(), 0, 2, 'id_product', 'ASC');
+        $results = [];
+
+        foreach ($products as $product) {
+            $p = new Product($product['id_product']);
+            $results[] = [
+                'text' => Product::getProductName($product['id_product']),
```

Comment:
Can't you get the product name since you instanciated in `$p`? `$p->name` ?
And you could rename this variable, either `$productEntity` or you can change your initial `$product` array to `$productData`

Reply:
Yes, `$p` was a temporary variable name, forgot to rename it...
I did try to use `$p->name` but it's an array with all the names for the different translations. Do you think I still should use it?

Reply:
Oh yes because it contains the localized name in an array No I think it's fine with what you did Just rename the variable it will be enough